src/random_test.cpp

```diff
@@ -1573,18 +1599,23 @@ Table *Table::table_id(TABLE_TYPES type, int id, Thd1 *thd) {
   }
 
   /* if temporary table encrypt variable set create encrypt table */
-
-  auto row = mysql_fetch_row(
-      get_result("select @@innodb_temp_tablespace_encrypt", thd));
+  auto result = get_result("select @@innodb_temp_tablespace_encrypt", thd);
+  if (! mysql_num_fields_safe(thd, result, 1)) {
+    throw std::runtime_error("Expected at least 1 field in result");
+  }
+  auto row = mysql_fetch_row_safe(thd, result);
   static std::string temp_table_encrypt = row[0];
 
   if (strcmp(FORK, "Percona-Server") == 0 && server_version() < 80000 &&
       temp_table_encrypt.compare("ON") == 0 && table->type == TEMPORARY)
     table->encryption = 'Y';
 
   /* if innodb system is encrypt , create encrypt table */
-  auto row1 = mysql_fetch_row(
-      get_result("select @@innodb_sys_tablespace_encrypt", thd));
+  result = get_result("select @@innodb_sys_tablespace_encrypt", thd);
```

Comment:
for MySQL this variables don't exists. and in that case, It throws and all other threads fall in infinite loop.